src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp

```diff
@@ -133,6 +135,109 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void EnableWindowsFeaturesDependencies(Execution::Context& context)
+    {
+        if (!Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::WindowsFeature))
+        {
+            return;
+        }
+
+        const auto& rootDependencies = context.Get<Execution::Data::Installer>()->Dependencies;
+
+        if (rootDependencies.Empty())
+        {
+            return;
+        }
+
+        if (rootDependencies.HasAnyOf(DependencyType::WindowsFeature))
+        {
+            context << Workflow::EnsureRunningAsAdmin;
+            if (context.IsTerminated())
+            {
+                return;
+            }
+
+            std::vector<std::string> invalidFeatures;
+
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&invalidFeatures](Dependency dependency)
+                {
+                    const auto& name = dependency.Id();
+                    WindowsFeature::WindowsFeature windowsFeature{ name };
+                    if (!windowsFeature.DoesExist())
+                    {
+                        invalidFeatures.emplace_back(name);
+                    }
+                });
+
+            if (!invalidFeatures.empty())
+            {
+                bool shouldTerminate = true;
+                auto warn = context.Reporter.Warn();
+                if (context.Args.Contains(Execution::Args::Type::IgnoreMissingDependencies))
+                {
+                    warn << Resource::String::WindowsFeatureNotFoundOverride << std::endl;
+                    shouldTerminate = false;
+                }
+                else
+                {
+                    warn << Resource::String::WindowsFeatureNotFoundOverrideRequired << std::endl;
+                }
+
+                warn << "  - " << Resource::String::WindowsFeaturesDependencies << std::endl;
+
+                for (const auto& feature : invalidFeatures)
+                {
+                    warn << "      " << feature << std::endl;
+                }
+
+                if (shouldTerminate)
+                {
+                    AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_INSTALL_MISSING_DEPENDENCY);
+                }
+            }
+
+            HRESULT hr = S_OK;
+            bool continueOnFailure = context.Args.Contains(Execution::Args::Type::Force);
+            bool rebootRequired = false;
+
+            context.Reporter.Info() << Resource::String::EnablingWindowsFeatures << std::endl;

```

Comment:
As a user, I would prefer this be shown per-feature as they are enabled; I don't need to be informed that features dependencies are being enabled in the case that all the features are already enabled